// @textlint/kernel/lib/types/src/index.js
"use strict";
/**
 * @fileoverview
 * Public function and interface should be included `Textlint` prefix or postfix.
 * It aim to avoid conflict naming on user land code.
 */
Object.defineProperty(exports, "__esModule", { value: true });
// SourceCode
var TextlintSourceCode_1 = require("./Source/TextlintSourceCode");
exports.TextlintSourceCode = TextlintSourceCode_1.TextlintSourceCode;
var TextlintRuleContextFixCommandGenerator_1 = require("./Rule/TextlintRuleContextFixCommandGenerator");
exports.TextlintRuleContextFixCommandGenerator = TextlintRuleContextFixCommandGenerator_1.TextlintRuleContextFixCommandGenerator;
var TextlintRuleError_1 = require("./Rule/TextlintRuleError");
exports.TextlintRuleError = TextlintRuleError_1.TextlintRuleError;
var TextlintRuleSeverityLevel_1 = require("./Rule/TextlintRuleSeverityLevel");
exports.TextlintRuleSeverityLevel = TextlintRuleSeverityLevel_1.TextlintRuleSeverityLevel;
// Rule
var TextlintRuleContext_1 = require("./Rule/TextlintRuleContext");
exports.TextlintRuleContext = TextlintRuleContext_1.TextlintRuleContext;
// Filter Rule
var TextlintFilterRuleContext_1 = require("./Rule/TextlintFilterRuleContext");
exports.TextlintFilterRuleContext = TextlintFilterRuleContext_1.TextlintFilterRuleContext;
// Plugin
var TextlintPluginModule_1 = require("./Plugin/TextlintPluginModule");
exports.TextlintPluginProcessor = TextlintPluginModule_1.TextlintPluginProcessor;
// Key Path
var TextlintKeyPath_1 = require("./KeyPath/TextlintKeyPath");
exports.normalizeTextlintPresetSubRuleKey = TextlintKeyPath_1.normalizeTextlintPresetSubRuleKey;
exports.normalizeTextlintRuleKey = TextlintKeyPath_1.normalizeTextlintRuleKey;
exports.normalizeTextlintRulePresetKey = TextlintKeyPath_1.normalizeTextlintRulePresetKey;
exports.normalizeTextlintPluginKey = TextlintKeyPath_1.normalizeTextlintPluginKey;
exports.normalizeTextlintFilterRuleKey = TextlintKeyPath_1.normalizeTextlintFilterRuleKey;
exports.normalizeTextlintKeyPath = TextlintKeyPath_1.normalizeTextlintKeyPath;
var TextlintPackagePrefix_1 = require("./KeyPath/TextlintPackagePrefix");
exports.TextlintPackageNamePrefix = TextlintPackagePrefix_1.TextlintPackageNamePrefix;
